bin/templates/cordova/lib/pluginHandlers.js

```diff
@@ -320,26 +320,33 @@ function generateAttributeError (attribute, element, id) {
 function getInstallDestination (obj) {
     var APP_MAIN_PREFIX = 'app/src/main';
 
-    if (obj.targetDir.startsWith('app')) {
+    var targetDirArray = obj.targetDir.split('/');
+
+    if (targetDirArray[0] === 'app') {
         // If any source file is using the new app directory structure,
         // don't penalize it
         return path.join(obj.targetDir, path.basename(obj.src));
-    } else if (obj.src.endsWith('.java')) {
-        return path.join(APP_MAIN_PREFIX, 'java', obj.targetDir.substring(4), path.basename(obj.src));
-    } else if (obj.src.endsWith('.aidl')) {
-        return path.join(APP_MAIN_PREFIX, 'aidl', obj.targetDir.substring(4), path.basename(obj.src));
-    } else if (obj.targetDir.includes('libs')) {
-        if (obj.src.endsWith('.so')) {
-            return path.join(APP_MAIN_PREFIX, 'jniLibs', obj.targetDir.substring(5), path.basename(obj.src));
-        } else {
+    } else {
```

Comment:
Why did you move the other conditions all into an else clause?

(I actually did some work to change it the other way.)

I think it does make the code a little more readable and more clear that the remaining conditions are based on the old directory structure. Makes it a little harder to review, though.

Reply:
I get it. I think the more logical and compartmentalized the code is, the more error prone and maintainable it is. Plus, when you will want to remove deprecated code you could safely remove the entire bloc. It is your call, I don't care, just need to make it work.

Reply:
I think you meant "less error prone", agreed on my part.

> Plus, when you will want to remove deprecated code you could safely remove the entire bloc

Yes and this function would probably be reduced to a single `path.join` operation, we could even get rid of it when we remove the deprecated code. This function was basically introduced with another name to deal with the deprecated "old fashioned" target directory scheme.

I would personally favor the way you proposed it, with a more clear comment as I requested before.